risc0/zkvm/src/host/server/exec/executor.rs

```diff
@@ -771,3 +785,58 @@ impl<'a> ExecutorImpl<'a> {
         ))
     }
 }
+
+#[derive(Debug, Serialize, Deserialize)]
+struct ExecutorStats {
+    execution_time: Duration,
+    segment_count: usize,
+    total_prover_cycles: u64,
+    user_instruction_cycles: u64,
+    cycles_used_ratio: f64,
+}
+
+impl ExecutorStats {
+    fn new(session: &Session, elapsed: Duration) -> Result<Self> {
+        let (total_prover_cycles, user_instruction_cycles) = session.get_cycles()?;
+        let cycles_used_ratio = user_instruction_cycles as f64 / total_prover_cycles as f64 * 100.0;
+        let segment_count = session.resolve()?.len();
+
+        Ok(Self {
+            total_prover_cycles,
+            user_instruction_cycles,
+            cycles_used_ratio,
+            segment_count,
+            execution_time: elapsed,
+        })
+    }
+
+    #[instrument(name = "executor", skip(self))]
+    fn log(&self) {
+        tracing::info!("execution time: {}", self.execution_time.human_duration());
+        tracing::info!(
+            "number of segments: {}",
+            self.segment_count.human_count_bare()
+        );
+        tracing::info!(
+            "total prover cycles: {}",
+            self.total_prover_cycles.human_count_bare()
+        );
+        tracing::info!(
+            "user instruction cycles: {}",
+            self.user_instruction_cycles.human_count_bare()
+        );
+        tracing::info!(
+            "available computation used: {}%",
+            self.cycles_used_ratio.human_count_bare()
+        );
```

Comment:
I thought it might be interesting to show users the total computation that the prover will undergo (considering continuations overhead - padding up to the nearest power of 2)  as well as the "real" computation we're executing. 

This would be the "global" value - for the whole session. 
Was thinking maybe in adding also the 50th, 75th, 95th and 99th percentile of "utilized computation" for the segments - so then we count how we're using or not each segment individually as well. 

Thoughts on this? 

(This might be out of the scope for this PR and I'm happy to open a new one for this if you see fit).

Reply:
(users will ask for help on how to squeeze their code to fit into a certain power of 2 but become desperate.